src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -98,6 +104,14 @@ protected Object readResolve() throws IOException {
         return this;
     }
 
+    /** Allows user to disable saving step info, if this imposes too high a performance overhead
+     *   by setting System Property org.jenkinsci.plugins.workflow.cps.DSL.keepStepInfo to 'false'.
+     *  This may be modified at runtime by the user if needed for debugging. */
+    private static boolean isKeepStepInfo() {
+        String prop = System.getProperty(DSL.class.getName()+".keepStepInfo");
```

Comment:
```
    public static String getProperty(String key) {
        checkKey(key);
        SecurityManager sm = getSecurityManager();
        if (sm != null) {
            sm.checkPropertyAccess(key);
        }

        return props.getProperty(key);
    }
```

vs static non-final field access for code that you have called out as performance critical.

Plus *not following existing patterns* which will significantly increase supportability costs.

For those two reasons I say this is decreasing the quality of the code which makes it a :bug: